File: src/main/java/org/example/kalkulationsprogramm/repository/BelegPositionRepository.java
package org.example.kalkulationsprogramm.repository;

import org.example.kalkulationsprogramm.domain.BelegPosition;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface BelegPositionRepository extends JpaRepository<BelegPosition, Long> {

    List<BelegPosition> findByBelegIdOrderBySortierungAsc(Long belegId);

    void deleteByBelegId(Long belegId);
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/BelegPositionRepository.kt
package org.example.kalkulationsprogramm.repository

import org.example.kalkulationsprogramm.domain.BelegPosition
import org.springframework.data.jpa.repository.JpaRepository

interface BelegPositionRepository : JpaRepository<BelegPosition, Long> {
    fun findByBelegIdOrderBySortierungAsc(belegId: Long?): List<BelegPosition>

    fun deleteByBelegId(belegId: Long?)
}
